src/services/uniswapV3/LiquidityService.ts

```diff
@@ -0,0 +1,415 @@
+
+    console.log('currentTickInitialized: ', currentTickInitialized);
+    const activeTickProcessed: TickProcessed = {
+      tickIdx: activeTickIdx,
+      liquidityActive: liquidity,
+      liquidityNet: JSBI.BigInt(0),
+      price0: parseFloat(tickToPrice(token0, token1, activeTickIdx).toSignificant(18)),
+      price1: parseFloat(tickToPrice(token1, token0, activeTickIdx).toSignificant(18)),
+      isCurrent: true,
+    };
+
+    const activeTick = tickIdxToTickDictionary[activeTickIdx];
+    if (activeTick) {
+      activeTickProcessed.liquidityNet = JSBI.BigInt(activeTick.liquidityNet);
+    }
+
+    const numSurroundingTicks = 100;
+
+    const createBarChartTicks = await this.createBarChartTicks(
+      activeTickIdx,
+      fullPool.liquidity,
+      tickSpacing,
+      fullPool.token0,
+      fullPool.token1,
+      numSurroundingTicks,
+      fullPool.fee,
+      graphTicks,
+    );
+
+    const poolSum = createBarChartTicks.reduce(
+      (acc, curr) => {
+        acc = {
+          liquidityActive: JSBI.toNumber(JSBI.add(JSBI.BigInt(acc.liquidityActive), JSBI.BigInt(curr.liquidityActive))),
+          liquidityLockedToken0: acc.liquidityLockedToken0 + curr.liquidityLockedToken0,
+          liquidityLockedToken1: acc.liquidityLockedToken1 + curr.liquidityLockedToken1,
+        };
+
+        return acc;
+      },
+      {liquidityActive: 0, liquidityLockedToken0: 0, liquidityLockedToken1: 0},
+    );
+
+    console.log('poolSum: ', poolSum);
+
+    return poolSum;
+  }
+
+  async createBarChartTicks(
+    tickCurrent: number,
+    poolLiquidity: JSBI,
+    tickSpacing: number,
+    token0: Token,
+    token1: Token,
+    numSurroundingTicks: number,
+    feeTier: FeeAmount,
+    graphTicks: GraphTick[],
+  ): Promise<BarChartTick[]> {
+    const processedTicks = this.processTicks(
+      tickCurrent,
+      poolLiquidity,
+      tickSpacing,
+      token0,
+      token1,
+      numSurroundingTicks,
+      graphTicks,
+    );
+
+    const barTicks = await Promise.all(
+      processedTicks.map(async (tick: TickProcessed) => {
+        return this.calculateLockedLiqudity(tick, token0, token1, tickSpacing, feeTier);
+      }),
+    );
+    barTicks.map((entry, i) => {
+      if (i > 0) {
+        barTicks[i - 1].liquidityLockedToken0 = entry.liquidityLockedToken0;
+        barTicks[i - 1].liquidityLockedToken1 = entry.liquidityLockedToken1;
+      }
+    });
+    return barTicks;
+  }
+
+  processTicks(
+    tickCurrent: number,
+    poolLiquidity: JSBI,
+    tickSpacing: number,
+    token0: Token,
+    token1: Token,
+    numSurroundingTicks: number,
+    graphTicks: GraphTick[],
+  ): TickProcessed[] {
+    const tickIdxToTickDictionary: Record<string, GraphTick> = Object.fromEntries(
+      graphTicks.map((graphTick) => [graphTick.tickIdx, graphTick]),
+    );
+
+    const liquidity = poolLiquidity;
+
+    let activeTickIdx = Math.floor(tickCurrent / tickSpacing) * tickSpacing;
```

Comment:
did you found this math somewhere in docs?

Reply:
please put this link to the comment of this line

Reply:
Done